src/components/application_manager/src/application_manager_impl.cc

```diff
@@ -3459,6 +3482,10 @@ event_engine::EventDispatcher& ApplicationManagerImpl::event_dispatcher() {
   return event_dispatcher_;
 }
 
+app_launch::AppLaunchCtrl& ApplicationManagerImpl::app_launch_ctrl() {
+  return *app_launch_ctrl_;
```

Comment:
@vlantonov, all objects which `ApplicationManagerImpl` is stores, are created within `ApplicationManagerImpl`. So, pointers to `HMICapabilities`, `ResumeCtrl` and `AppLaunchCtrl` are stored inside of it, and initialized by `ApplicationManagerImpl` inside its constructor. And there is no chance that the value of this pointers can be changed from the outside of `ApplicationManagerImpl`.